src/frequenz/sdk/timeseries/formula_engine/_formula_steps.py

```diff
@@ -529,7 +529,15 @@ async def _fetch_next(self) -> Sample[QuantityT] | None:
         try:
             next_value = await self._stream.receive()
         except ReceiverError[Any] as err:
```

Comment:
My bet would be that `ReceiverError[Any]` is not really a `ReceiverError` but some `type` object or something like that. Maybe try replacing by just `ReceiverError`?

I just did a quick test and it is that:

```py
from typing import Any, Generic, TypeVar

T = TypeVar("T")

class Ex(Exception, Generic[T]):
    pass

try:
    raise Ex("foo")
except Ex[Any] as e:
    print(e)
```

This triggers the same issue, removing the `[Any]` makes it work.

Reply:
You are right! :) It was ReceiverStoppedError .

Reply:
What I mean is the generic type specification was wrong, a `ReceiverStoppedError` is still a `ReceiverError`, but you can't use the type parameter in `except` clauses.